rules/TypeDeclaration/NodeAnalyzer/ReturnTypeAnalyzer/StrictScalarReturnTypeAnalyzer.php

```diff
@@ -46,4 +55,25 @@ public function matchAlwaysScalarReturnType(ClassMethod|Closure|Function_ $funct
 
         return $this->typeFactory->createMixedPassedOrUnionType($scalarTypes);
     }
+
+    private function isHardCodedExpression(Expr $expr): bool
+    {
+        // Normal scalar values like strings, integers and floats
+        if ($expr instanceof Scalar) {
+            return true;
+        }
+
+        // true / false / null are constants
+        if ($expr instanceof ConstFetch &&
+            in_array($expr->name->toLowerString(), ['true', 'false', 'null'], true)) {
+            return true;
+        }
+
+        // Negative numbers are wrapped in UnaryMinus, so check expression inside it
+        if ($expr instanceof UnaryMinus && $expr->expr instanceof Scalar) {
```

Comment:
How about UnaryPlus?

Reply:
That was an oversight / never have I written `+1` instead of just `1`. So didn't test for it either (so test added and added it here as well).